### PR TITLE
Fixup Username semantics and implementation

### PR DESCRIPTION
**Pull request by decentralion**
This commit incorporates post-merge review feedback from @wchargin,
as found in [this review].

[this review]: https://github.com/sourcecred/sourcecred/pull/1915#pullrequestreview-442543321

Test plan: Unit tests updated; `yarn test` passes.

### CODE REVIEW

File: src/ledger/user.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ import {
  * We validate usernames using GitHub-esque rules.
  */
 export opaque type Username: string = string;
-const USERNAME_PATTERN = /^@?([A-Za-z0-9-_]+)$/;
+const USERNAME_PATTERN = /^[A-Za-z0-9-]+$/;
+const _USERNAME_REGEXP = new RegExp(USERNAME_PATTERN);
```
[the remark this answers is not in the document]
Comment (decentralion) [reply src/ledger/user.js:43]: ...i was cargo culting... fixed.

File: src/ledger/user.js [hunk 2]
```diff
@@ -73,12 +78,11 @@ export function createUser(name: string): User {
...
  */
 export function usernameFromString(username: string): Username {
-  const re = new RegExp(USERNAME_PATTERN);
-  const match = re.exec(username);
+  const match = username.match(_USERNAME_REGEXP);
   if (match == null) {
     throw new Error(`invalid username: ${username}`);
   }
-  return match[1];
+  return match[0];
```
[the remark this answers is not in the document]
Comment (decentralion) [reply src/ledger/user.js:85]: Done.

**Comment by decentralion** [comment]
Took all your suggestions, including making usernames all canonically lower case, which will surely save some confusion down the line. Thanks!

**Comment by wchargin** [comment]
> which will surely save some confusion down the line. Thanks!

Indeed; `decentraIion` homonym attack averted for now. :-)

**Pull request merged**